### PR TITLE
Refactor getRecommendationsUrl and enable `unirecs_datalab` engine for 3 services

### PR DESCRIPTION
**Pull request by mfonofm**
**Overall change:**
Refactoring `getRecommendationsUrl` to simplify and enable new recommendations endpoint. Adds checks to enable `unirecs_datalab` engine for the Mundo, Indonesian and Turkish services

**Code changes:**

- remove `portugueseRecommendationsExperimentEndpoint` 
- refactor `getRecommendationsUrl`
- update `getRecommendationsUrl` tests

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/utilities/getUrlHelpers/getRecommendationsUrl/index.js [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
-    ? `${process.env.RECOMMENDATIONS_ENDPOINT}/recommendations${assetUri}?Engine=${engine}&EngineVariant=${engineVariant}`
-    : `${process.env.RECOMMENDATIONS_ENDPOINT}/recommendations${assetUri}?Engine=${engine}`;
+export default ({ assetUri, engine, engineVariant }) => {
+  if (engineVariant) {
+    return `${process.env.RECOMMENDATIONS_ENDPOINT}/recommendations${assetUri}?Engine=${engine}&EngineVariant=${engineVariant}`;
+  }
+  if (engine) {
+    return `${process.env.RECOMMENDATIONS_ENDPOINT}/recommendations${assetUri}?Engine=${engine}`;
+  }
+  return `${process.env.RECOMMENDATIONS_ENDPOINT}/recommendations${assetUri}`;
```
Comment (andrewscfc) [inline src/app/lib/utilities/getUrlHelpers/getRecommendationsUrl/index.js]: This just makes sure it doesn't provide an `EngineVariant` if no `Engine` is provided as it doesn't make a lot of sense. That shouldn't happen though really.

```suggestion
  let queryParams = '';
  if (engine) {
    queryParams += `Engine=${engine}`;
    if(engineVariant) {
      queryParams += `&EngineVariant=${engineVariant`;
    }
  }
  return `${process.env.RECOMMENDATIONS_ENDPOINT}/recommendations${assetUri}${queryParams}`;
```

File: src/app/routes/cpsAsset/utils/getAdditionalPageData.js [hunk 2]
```diff
@@ -31,7 +29,7 @@ const getRecommendations = (service, assetUri) => {
       {
         name: 'recommendations',
         attachAgent: true,
-        path: getRecommendationsUrl({ assetUri }),
+        path: getRecommendationsUrl({ assetUri, engine: 'unirecs_datalab' }),
```
Comment (andrewscfc) [inline src/app/routes/cpsAsset/utils/getAdditionalPageData.js]: Are there any services left still using camino? If so we need to add a conditional for those that are using unirecs and those that are still using camino?

Comment (JonBeeb) [reply src/app/routes/cpsAsset/utils/getAdditionalPageData.js]: Yes. I think we have 3 (off top of my head).

**Comment by amoore108** [comment]
Do we need to have 2 instances of the recommendations for all the services apart from `portuguese`?:
https://github.com/bbc/simorgh/blob/7c0ffe86664f9744781e6ba0c53f1f67d0e93669/src/app/routes/cpsAsset/utils/getAdditionalPageData.js#L28-L50

Would something like this work for ensuring only the 3 services in tickets are using the `unirecs_datalab` engine?:
```javascript
  if (service !== 'portuguese' || isLive()) {
    const UNIRECS_ALLOW_LIST = ['indonesia', 'mundo', 'turkce'];

    return [
      {
        name: 'recommendations',
        attachAgent: true,
        path: getRecommendationsUrl({
          assetUri,
          ...(UNIRECS_ALLOW_LIST.includes(service) && {
            engine: 'unirecs_datalab',
          }),
        }),
        assetUri,
        api: 'recommendations',
        apiContext: 'secondary_data',
      },
    ];
  }
```

**Comment by HarveyPeachey** [comment]
> Do we need to have 2 instances of the recommendations for all the services apart from `portuguese`?:
> 
> https://github.com/bbc/simorgh/blob/7c0ffe86664f9744781e6ba0c53f1f67d0e93669/src/app/routes/cpsAsset/utils/getAdditionalPageData.js#L28-L50
> 
> Would something like this work for ensuring only the 3 services in tickets are using the `unirecs_datalab` engine?:
> 
> ```js
>   if (service !== 'portuguese' || isLive()) {
>     const UNIRECS_ALLOW_LIST = ['indonesia', 'mundo', 'turkce'];
> 
>     return [
>       {
>         name: 'recommendations',
>         attachAgent: true,
>         path: getRecommendationsUrl({
>           assetUri,
>           ...(UNIRECS_ALLOW_LIST.includes(service) && {
>             engine: 'unirecs_datalab',
>           }),
>         }),
>         assetUri,
>         api: 'recommendations',
>         apiContext: 'secondary_data',
>       },
>     ];
>   }
> ```

That looks like the right approach

**Review by andrewscfc — APPROVED** [review]

**Review by amoore108 — APPROVED** [review]

**Pull request merged**